Get Recent Media

aahiltn opened the pull request:
## Description
Allows the user to get the most recent songs/albums. Songs are currently prioritized over albums. 

Changes :  
- Added a cover photo and release date to the Track struct. Not sure if this was the most efficient way to hold these fields, but I also changed the current search by name to also return these fields for tracks.
- Uses query params. Most reviews would likely want to use these as well, maybe /media/?sort=reviews

[Link to Ticket](https://github.com/GenerateNU/platnm/issues/33)

## How Has This Been Tested?
As advised, see Postman screenshots below.

## Checklist:
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] Some tests are included that test my code

## Screenshots:
<!--- If working on a backend ticket, screenshots or a walkthrough of successful API calls are included. -->
Sort by Recent:
![image](https://github.com/user-attachments/assets/3fd3bced-e7c8-473b-bb2b-9ca1799e3698)

Search by Name (now with photo & date):
![image](https://github.com/user-attachments/assets/d9475743-2818-47f1-9fae-7362da76c05c)

closes #33

ddusichka commented on backend/internal/models/media.go:
```diff
@@ -28,11 +28,13 @@ type Album struct {
-	AlbumID  int    `json:"album_id"`
-	Title    string `json:"title"`
-	Duration int    `json:"duration_seconds"`
+	Media       MediaType
```
```suggestion
	Media       MediaType `json:"media"`
```
super small but can we give this a new json tag to match the rest

ddusichka commented on backend/internal/service/handler/media/get_media.go:
```diff
@@ -9,3 +10,19 @@ func (h *Handler) GetMediaByName(c *fiber.Ctx) error {
+func (h *Handler) GetMedia(c *fiber.Ctx) error {
+	// Check for the "type" query parameter
+	mediaType := c.Query("sort")
+	fmt.Println(mediaType)
```
remove print :)

ddusichka commented on backend/internal/service/handler/media/get_media.go:
```diff
@@ -9,3 +10,19 @@ func (h *Handler) GetMediaByName(c *fiber.Ctx) error {
+		return c.Status(fiber.StatusOK).JSON(medias)
+	}
+
+	return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid type query parameter"})
```
I think we might want to handle this case differently (return _some_ media rather than throw), since query parameters should be treated as optional and not require an error if invalid or not provided. Right now though I'm not totally sure which media makes the most sense to return so this can be a future ticket!

aahiltn replied:
Yeah, the logic that I was thinking would be to return "recommended media" if that's something we were doing? If not, maybe media that was shared with the user could be returned by this endpoint? The error was a temporary fix of course.

ddusichka replied:
these are both great ideas i'm taking notes 📝

ddusichka commented on backend/internal/storage/postgres/schema/media.go:
```diff
@@ -11,11 +13,122 @@ type MediaRepository struct {
 	// select all rows where either the input string is in the title of the media, or if the string matches one of the titles fuzzily
 	var albumQuery = "SELECT * FROM album WHERE levenshtein(title, $1) <= 5 OR title ILIKE '%' || $1 || '%' LIMIT 20;"
-	var trackQuery = "SELECT * FROM track WHERE levenshtein(title, $1) <= 5 OR title ILIKE '%' || $1 || '%' LIMIT 20;"
+	var trackQuery = `
```
thank u for fixing this query!

ddusichka commented on backend/internal/models/media.go:
```diff
@@ -28,11 +28,13 @@ type Album struct {
-	Duration int    `json:"duration_seconds"`
+	Media       MediaType
+	ID          int       `json:"id"`
+	AlbumID     int       `json:"album_id"`
```
lowkey can we add and populate AlbumName in here? I think we should keep the ID too so we can do more queries if needed but having the name for quick display would be nice

ddusichka replied:
I think we should also add artist names (and IDs) to both track and album, but we can save that for later since the multiple artist edge case makes the data kinda weird

ddusichka reviewed (commented):
looks good, few small suggestions!

aahiltn commented:
Added AlbumTitle to the Track struct, and updated both endpoints!
By date:
![image](https://github.com/user-attachments/assets/bcffebc5-6730-4781-8ffe-d91f0f11d52e)

By name: 
![image](https://github.com/user-attachments/assets/9a50a0de-fc53-4523-b2ed-192a86180104)

aahiltn commented:
Also just fixed the capitals being off, whoops. Track and Album now have the same json tag. 
![image](https://github.com/user-attachments/assets/683966db-3fde-4e6a-abab-8205a1a7d160)

ddusichka reviewed (approved):
yay! looks good to me
